assets/javascripts/discourse/initializers/add-search-results.js.es6

```diff
@@ -0,0 +1,112 @@
+import { ajax } from "discourse/lib/ajax";
+import { withPluginApi } from "discourse/lib/plugin-api";
+import Post from "discourse/models/post";
+import Topic from "discourse/models/topic";
+import {
+  ENCRYPT_ACTIVE,
+  getEncryptionStatus,
+  getIdentity,
+} from "discourse/plugins/discourse-encrypt/lib/discourse";
+import {
+  decrypt,
+  importKey,
+} from "discourse/plugins/discourse-encrypt/lib/protocol";
+import I18n from "I18n";
+
+const CACHE_KEY = "discourse-encrypt-cache";
+
+function getCache() {
+  const cache = window.sessionStorage.getItem(CACHE_KEY);
+  return cache ? JSON.parse(cache) : {};
+}
+
+function addCacheItem(type, item) {
+  const cache = getCache();
+  if (!cache[type]) {
+    cache[type] = [];
+  } else if (item.id) {
+    cache[type] = cache[type].filter((i) => i.id !== item.id);
+  }
+  cache[type].push(item);
+  window.sessionStorage.setItem(CACHE_KEY, JSON.stringify(cache));
+}
+
+export default {
+  name: "add-search-results",
+
+  initialize(container) {
+    const currentUser = container.lookup("current-user:main");
+    const siteSettings = container.lookup("site-settings:main");
+    if (getEncryptionStatus(currentUser, siteSettings) !== ENCRYPT_ACTIVE) {
+      return;
+    }
+
+    if (!window.sessionStorage.getItem(CACHE_KEY)) {
+      ajax("/encrypt/posts").then((result) => {
```

Comment:
This happens on every empty cache app boot, regardless of whether the user searches. Could we make it so that it only performs this when the user goes to a search?

Reply:
I hesitated adding the hook on search because it would make searching slower (especially for users with lots of topics) and we'd need some promise based API. I made the changes, but I still have to do some testing and see the time difference.